CalendarApp/GoogleCalendarService/Manager/EventsListBuilder.cs

```diff
@@ -0,0 +1,254 @@
+                case "MONTHLY": _frequencyEnumaretion = FrequencyEnumaretion.MONTHLY; break;
+                case "YEARLY": _frequencyEnumaretion = FrequencyEnumaretion.YEARLY; break;
+            }
+
+            return this;
+        }
+
+        public EventsListBuilder SetSingleEvent(Event realEvent)
+        {
+            _freqType = FrequencyType.COUNT;
+            _count = 1;
+            _frequencyEnumaretion = FrequencyEnumaretion.DAILY;
+            return this;
+        }
+
+        public EventsListBuilder SetRealEvent(Event realEvent)
+        {
+            _realEvent = realEvent;
+            return this;
+        }
+
+
+        public EventsListBuilder SetCount(string count)
+        {
+            if (count != "")
+            {
+                _freqType = FrequencyType.COUNT;
+                _count = Convert.ToInt32(count);
+            }
+
+            return this;
+        }
+
+        public EventsListBuilder SetUntil(string until)
+        {
+            if (until != "")
+            {
+                _freqType = FrequencyType.UNTIL;
+                _untilDateTime = GetUntilDateTime(until);
+            }
+
+            return this;
+        }
+
+        public EventsListBuilder SetByDay(string byDay)
+        {
+            if(byDay!= "")
+                SetDayOfWeeksList(byDay);
+            return this;
+        }
+
+        private void SetDayOfWeeksList(string byDay)
+        {
+            _dayOfWeeksList = byDay.Split(',').Select(ToDayOfWeek).ToList();
+            _dayJumpList = GetDayJumpsList();
+        }
+
+        private DayOfWeek ToDayOfWeek(string day)
+        {
+            switch (day)
+            {
+                case "SU":
+                    return DayOfWeek.Sunday;
+                case "MO":
+                    return DayOfWeek.Monday;
+                case "TU":
+                    return DayOfWeek.Tuesday;
+                case "WE":
+                    return DayOfWeek.Wednesday;
+                case "TH":
+                    return DayOfWeek.Thursday;
+                case "FR":
+                    return DayOfWeek.Friday;
+                case "SA":
+                    return DayOfWeek.Saturday;
+                default:
+                    return DayOfWeek.Sunday;
+            }
+        }
+
+        private bool CheckForCrossBreak(FrequencyType freqType, int i = 0, int count = 0, DateTime startTime = default(DateTime), DateTime untilDateTine = default(DateTime))
+        {
+            switch (freqType)
+            {
+                case FrequencyType.COUNT:
+                {
+                    if (i == count - 1)
+                        return true;
+                }
+                    break;
+
+                case FrequencyType.UNTIL:
+                {
+                    if(startTime.CompareTo(untilDateTine) > 0)
+                    return true;
+                }
+                    break;
+            }
+
+            return false;
+        }
```

Comment:
Can this method be refactored? I'm not crazy about all those optional parameters.